cns/requestcontroller/kubecontroller/crdrequestcontroller.go

```diff
@@ -42,6 +43,9 @@ type crdRequestController struct {
 	CNSClient       cnsclient.APIClient
 	nodeName        string //name of node running this program
 	Reconciler      *CrdReconciler
+	initialized     bool
+	Started         bool
+	lock            sync.Mutex
```

Comment:
After looking at this and its usage for a bit, I understand the intent, but I think this is a sign that this struct needs to be split into different abstractions. Ideally, this shouldn't need to carry state about initialization, we should just have ready-to-go dependencies injected, or document proper usage. I would suggest adding a todo to clean up further because the intention of these is hard to follow. Also, since there is a getter, `Started` need not be exported.

Reply:
This piece of code requires adding more tests. Created a PBI to incorporate feedback and add more tests

---
In reply to: [585726500](https://github.com/Azure/azure-container-networking/pull/809#discussion_r585726500) [](ancestors = 585726500)